core/src/main/scala/jms4s/jms/JmsContext.scala

```diff
@@ -0,0 +1,79 @@
+package jms4s.jms
+
+import cats.effect.{ Blocker, Concurrent, ContextShift, Resource, Sync }
+import cats.implicits._
+import io.chrisdavenport.log4cats.Logger
+import javax.jms.JMSContext
+import jms4s.config.{ DestinationName, QueueName, TopicName }
+import jms4s.jms.JmsDestination.{ JmsQueue, JmsTopic }
+import jms4s.jms.JmsMessage.JmsTextMessage
+import jms4s.model.SessionType
+
+import scala.concurrent.duration.FiniteDuration
+
+class JmsContext[F[_]: Sync: Logger: ContextShift: Concurrent](
+  private val context: JMSContext,
+  private[jms4s] val blocker: Blocker
+) {
+
+  def createContext(sessionType: SessionType): Resource[F, JmsContext[F]] =
+    Resource
+      .make(
+        Logger[F].info("Creating context") *>
+          blocker.delay(context.createContext(sessionType.rawAcknowledgeMode))
+      )(
+        context =>
+          Logger[F].info("Releasing context") *>
```

Comment:
I'd also add the `context` in the log (e.g. `Logger[F].info(s"Releasing context $context...")`) since it may be useful for troubleshooting.